modules/netdevedit.php

```diff
@@ -573,7 +573,7 @@
         $macs = array();
         foreach ($nodeipdata['macs'] as $key => $value) {
             if (check_mac($value)) {
-                if ($value != '00:00:00:00:00:00' && !ConfigHelper::checkValue(ConfigHelper::getConfig('phpui.allow_mac_sharing', true))) {
+                if ($value != '00:00:00:00:00:00' && !ConfigHelper::checkConfig('phpui.allow_mac_sharing')) {
                     if (($nodeid = $LMS->GetNodeIDByMAC($value)) != null && $nodeid != $_GET['ip']) {
```

Comment:
To zmieni zachowanie, gdyż checkConfig() domyślnie zakłada false.

Reply:
Domyślnie w LMS allow_mac_sharing jest false 
(https://github.com/chilek/lms-plus/blob/master/doc/lms.pgsql#L3309).

Są 3 przypadki:
- zmienna nie istnieje
- zmienna istnieje i jest false
- zmienna istnieje i jest true

Tylko w ostatnim przypadku wykonuje się kod (tak jak w pozostałych formularzach). I jest to słuszne zachowanie.

Reply:
Z jakiegoś powodu tu było _true_ i pewnie nie było to przypadkowe.

Reply:
Gdy nie ma zmiennej możesz zedytować mac na dublet (ominąć domyślne zachowanie). Obecnie w kodzie jest błąd.

Reply:
To chyba faktycznie błąd wprowadzony commitem:
```diff
commit 2e707ce7bdd57e6f595545f4779389e04452c94d
Author: Maciej Lew <maciej.lew.1987@gmail.com>
Date:   Thu Jul 10 21:31:43 2014 +0200

    LMSConfig integration
    
    Fixes some bugs in LMSConfig.
    Adds user priviledges integration to LMSConfig.
    Replaces check_conf and get_conf functions.

diff --git a/modules/netdevedit.php b/modules/netdevedit.php
index 4484d2027..598747a98 100644
--- a/modules/netdevedit.php
+++ b/modules/netdevedit.php
@@ -398,7 +398,7 @@ switch ($action) {
                $macs = array();
                foreach ($nodeipdata['macs'] as $key => $value)
                        if (check_mac($value)) {
-                               if ($value != '00:00:00:00:00:00' && !chkconfig($CONFIG['phpui']['allow_mac_sharing']))
+                               if ($value != '00:00:00:00:00:00' && !ConfigHelper::checkValue(ConfigHelper::getConfig('phpui.allow_mac_sharing', false)))
                                        if ($LMS->GetNodeIDByMAC($value))
                                                $error['mac' . $key] = trans('MAC address is in use!');
                                $macs[] = $value;
@@ -492,7 +492,7 @@ switch ($action) {
                $macs = array();
                foreach ($nodeipdata['macs'] as $key => $value)
                        if (check_mac($value)) {
-                               if ($value != '00:00:00:00:00:00' && isset($CONFIG['phpui']['allow_mac_sharing']) && !chkconfig($CONFIG['phpui']['allow_mac_sharing']))
+                               if ($value != '00:00:00:00:00:00' && !ConfigHelper::checkValue(ConfigHelper::getConfig('phpui.allow_mac_sharing', true)))
                                        if (($nodeid = $LMS->GetNodeIDByMAC($value)) != NULL && $nodeid != $_GET['ip'])
                                                $error['mac' . $key] = trans('MAC address is in use!');
                                $macs[] = $value;
@@ -655,7 +655,7 @@ include(MODULES_DIR . '/netdevxajax.inc.php');
 
 switch ($edit) {
        case 'data':
-               if (chkconfig($CONFIG['phpui']['ewx_support']))
+               if (ConfigHelper::checkValue(ConfigHelper::getConfig('phpui.ewx_support', false)))
                        $SMARTY->assign('channels', $DB->GetAll('SELECT id, name FROM ewx_channels ORDER BY name'));
 
                $SMARTY->display('netdevedit.html');
```

Reply:
Zdarza się ;-) każdy popełnia błędy. To nic newralgicznego.